dbt/adapters/athena/impl.py

```diff
@@ -57,6 +57,48 @@ def convert_number_type(cls, agate_table: agate.Table, col_idx: int) -> str:
     def convert_datetime_type(cls, agate_table: agate.Table, col_idx: int) -> str:
         return "timestamp"
 
+    # TODO: Add more lf-tag unit tests when moto supports lakeformation
+    # moto issue: https://github.com/getmoto/moto/issues/5964
+    @available
+    def add_lf_tags(self, database: str, table: str = None, lf_tags: Dict[str, str] = None):
+        conn = self.connections.get_thread_connection()
+        client = conn.handle
+
+        lf_tags = lf_tags or conn.credentials.lf_tags
+        if not lf_tags:
+            logger.debug("No LF tags configured")
+            return
+
+        resource = {
+            "Database": {"Name": database},
+        }
+
+        if table:
+            resource = {
+                "Table": {
+                    "DatabaseName": database,
+                    "Name": table,
+                }
+            }
+
+        with boto3_client_lock:
+            lf_client = client.session.client(
+                "lakeformation", region_name=client.region_name, config=get_boto3_config()
+            )
+
+        lf_client.add_lf_tags_to_resource(
+            Resource=resource,
+            LFTags=[
+                {
+                    "TagKey": key,
+                    "TagValues": [
+                        value,
+                    ],
+                }
+                for key, value in lf_tags.items()
+            ],
+        )
+
```

Comment:
let's add a little logger.debug here `logger.debug(f'{lf_tags} added to {database}.{table}')` or something similar, helpfull to debug.